Address bug 34589558 "BIND 9.16.29 named socket.c:3459: fatal error:
open(/dev/poll) failed: Too many open files".

I shall inform ISC of this need to patch in due course.

diff -r 598de125ad73 bin/named/unix/os.c
--- a/bin/named/unix/os.c	Thu Sep 15 07:02:36 2022 -0700
+++ b/bin/named/unix/os.c	Thu Sep 15 07:10:27 2022 -0700
@@ -485,13 +485,15 @@
 
 void
 named_os_adjustnofile(void) {
-#if defined(__linux__)
+#if defined(__linux__) || defined(__sun__)
 	isc_result_t result;
 	isc_resourcevalue_t newvalue;
 
 	/*
 	 * Linux: max number of open files specified by one thread doesn't seem
 	 * to apply to other threads on Linux.
+	 * Sun: restriction needs to be removed sooner when hundreds of CPUs
+	 * are available.
 	 */
 	newvalue = ISC_RESOURCE_UNLIMITED;
 
@@ -499,7 +501,7 @@
 	if (result != ISC_R_SUCCESS) {
 		named_main_earlywarning("couldn't adjust limit on open files");
 	}
-#endif /* if defined(__linux__) */
+#endif /* if defined(__linux__) || defined(__sun__) */
 }
 
 void
